src/Numerics/DGMethods/DGModel.jl

```diff
@@ -1043,3 +870,575 @@ function continuous_field_gradient!(
     )
     wait(device, event)
 end
+
+function hyperdiff_indexmap(balance_law, ::Type{FT}) where {FT}
+    ns_hyperdiff = number_states(balance_law, Hyperdiffusive())
+    if ns_hyperdiff > 0
+        return varsindices(
+            vars_state(balance_law, Gradient(), FT),
+            fieldnames(vars_state(balance_law, GradientLaplacian(), FT)),
+        )
+    else
+        return nothing
+    end
+end
+
+function basic_launch_info(dg::DGModel)
```

Comment:
might want to combine this with `basic_grid_info` defined at L60, looks like they are doing most of the same things

Reply:
Yeah, at first I wanted to use `basic_grid_info` but it then I realized it assumes stacked grids. Of course, this could be fixed by having two methods of `basic_grid_info`. I wasn't sure how much of what need to launch kernels with different polynomial order will be truly basic grid information so I added `basic_launch_info` for now.